### PR TITLE
Implementation of revised version of DNSSEC01

### PR DESCRIPTION
**Pull request by vlevigneron**
Fixes zonemaster/zonemaster-engine#616
Fixes zonemaster/zonemaster-engine#308

### CODE REVIEW

File: lib/Zonemaster/Engine/Test/DNSSEC.pm [hunk 1]
```diff
@@ -486,18 +501,30 @@ Readonly my %TAG_DESCRIPTIONS => (
         __x    # DNSKEY_SIGNED
           "The apex DNSKEY RRset was correcly signed.", @_;
     },
+    DS_ALGORITHM_NOT_DS => sub {
+        __x    # DS_ALGORITHM_NOT_DS
+          "{ns}/{address} answered DS query for zone {zone} with a reserved algorithm ({algorithm_number}) for keytag {keytag}.", @_;
+    },
+    DS_ALGORITHM_DEPRECATED => sub {
+        __x    # DS_ALGORITHM_DEPRECATED
+          "{ns}/{address} answered DS query for zone {zone} with a deprecated algorithm ({algorithm_number}/{algorithm_mnemonic}) for keytag {keytag}.", @_;
```
Comment (matsduf) [inline lib/Zonemaster/Engine/Test/DNSSEC.pm]: I suggest:
```
{ns}/{address} returned a DS record created by algorithm ({algorithm_number}/{algorithm_mnemonic}), which is deprecated. The DS record is for the DNSKEY record with keytag {keytag} in zone {zone}.
```

File: lib/Zonemaster/Engine/Test/DNSSEC.pm [hunk 2]
```diff
@@ -486,18 +501,30 @@ Readonly my %TAG_DESCRIPTIONS => (
...
+    },
+    DS_ALGORITHM_MISSING => sub {
+        __x    # DS_ALGORITHM_MISSING
+          "{ns}/{address} answered DS query for zone {zone} with NO DS record for keytag {keytag} using algorithm {algorithm_number}/{algorithm_mnemonic}, which is not OK.", @_;
```
Comment (matsduf) [inline lib/Zonemaster/Engine/Test/DNSSEC.pm]: I suggest:
```
{ns}/{address} returned no DS record created by algorithm {algorithm_number}/{algorithm_mnemonic} for zone {zone}, which is required.
```

File: lib/Zonemaster/Engine/Test/DNSSEC.pm [hunk 3]
```diff
@@ -486,18 +501,30 @@ Readonly my %TAG_DESCRIPTIONS => (
...
+    },
+    DS_ALGORITHM_OK => sub {
+        __x    # DS_ALGORITHM_OK
+          "{ns}/{address} answered DS query for zone {zone} with DS record for keytag {keytag} using algorithm {algorithm_number}/{algorithm_mnemonic}, which is OK.", @_;
```
Comment (matsduf) [inline lib/Zonemaster/Engine/Test/DNSSEC.pm]: I suggest:
```
{ns}/{address} returned a DS record created by algorithm {algorithm_number}/{algorithm_mnemonic}, which is OK. The DS record is for the DNSKEY record with keytag {keytag} in zone {zone}.
```

File: lib/Zonemaster/Engine/Test/DNSSEC.pm [hunk 4]
```diff
@@ -486,18 +501,30 @@ Readonly my %TAG_DESCRIPTIONS => (
...
+    },
+    DS_ALGORITHM_RESERVED => sub {
+        __x    # DS_ALGORITHM_RESERVED
+          "{ns}/{address} answered DS query for zone {zone} with an unassigned algorithm ({algorithm_number}) for keytag {keytag}.", @_;
```
Comment (matsduf) [inline lib/Zonemaster/Engine/Test/DNSSEC.pm]: I suggest:
```
{ns}/{address} returned a DS record created by with an unassigned algorithm ({algorithm_number}), which is not OK. The DS record is for the DNSKEY record with keytag {keytag} in zone {zone}.
```

File: lib/Zonemaster/Engine/Test/DNSSEC.pm [hunk 5]
```diff
@@ -718,6 +749,10 @@ Readonly my %TAG_DESCRIPTIONS => (
         __x    # TOO_MANY_ITERATIONS
           "The number of NSEC3 iterations is {count}, which is too high for key length {keylength}.", @_;
     },
+    UNEXPECTED_RESPONSE_DS => sub {
+        __x    # UNEXPECTED_RESPONSE_DS
+          "{ns}/{address} answered DS query for zone {zone} with an unexpected rcode ({rcode}).", @_;
```
Comment (matsduf) [inline lib/Zonemaster/Engine/Test/DNSSEC.pm]: I suggest:
```
{ns}/{address} responded with an unexpected rcode ({rcode}) on a DS query for zone {zone} .
```

**Comment by vlevigneron** [comment]
@matsduf I agree we should not update all .po files each time, in particular while @mattias-p has not finished and merged his work on translation and gettext issues.That said, that's why I only update en.po to be sure that new messages are all there and old ones removed. For the next PR, if everybody agrees, we can stop to update .po files.

File: lib/Zonemaster/Engine/Test/DNSSEC.pm [hunk 6]
```diff
@@ -486,18 +501,30 @@ Readonly my %TAG_DESCRIPTIONS => (
         __x    # DNSKEY_SIGNED
           "The apex DNSKEY RRset was correcly signed.", @_;
     },
+    DS_ALGORITHM_NOT_DS => sub {
+        __x    # DS_ALGORITHM_NOT_DS
+          "{ns}/{address} answered DS query for zone {zone} with a reserved algorithm ({algorithm_number}) for keytag {keytag}.", @_;
+    },
```
[the remark this answers is not in the document]
Comment (vlevigneron) [reply lib/Zonemaster/Engine/Test/DNSSEC.pm]: Modified in https://github.com/zonemaster/zonemaster-engine/pull/627/commits/dc8edf852cb796eebe65893f2595c277e9c6512a

File: lib/Zonemaster/Engine/Test/DNSSEC.pm (hunk 1 above)
Comment (vlevigneron) [reply lib/Zonemaster/Engine/Test/DNSSEC.pm]: Modified in https://github.com/zonemaster/zonemaster-engine/pull/627/commits/dc8edf852cb796eebe65893f2595c277e9c6512a

File: lib/Zonemaster/Engine/Test/DNSSEC.pm (hunk 2 above)
Comment (vlevigneron) [reply lib/Zonemaster/Engine/Test/DNSSEC.pm]: Modified in https://github.com/zonemaster/zonemaster-engine/pull/627/commits/dc8edf852cb796eebe65893f2595c277e9c6512a

File: lib/Zonemaster/Engine/Test/DNSSEC.pm (hunk 3 above)
Comment (vlevigneron) [reply lib/Zonemaster/Engine/Test/DNSSEC.pm]: Modified in https://github.com/zonemaster/zonemaster-engine/pull/627/commits/dc8edf852cb796eebe65893f2595c277e9c6512a

File: lib/Zonemaster/Engine/Test/DNSSEC.pm (hunk 4 above)
Comment (vlevigneron) [reply lib/Zonemaster/Engine/Test/DNSSEC.pm]: Modified in https://github.com/zonemaster/zonemaster-engine/pull/627/commits/dc8edf852cb796eebe65893f2595c277e9c6512a

File: lib/Zonemaster/Engine/Test/DNSSEC.pm (hunk 5 above)
Comment (vlevigneron) [reply lib/Zonemaster/Engine/Test/DNSSEC.pm]: Modified in https://github.com/zonemaster/zonemaster-engine/pull/627/commits/dc8edf852cb796eebe65893f2595c277e9c6512a

**Pull request merged**